### ISSUE TITLE
Ubuntu 18 GDB issue. Truncated register 26 in remote 'g' packet

### ISSUE THREAD

**Issue by user000543** [qilingframework/qiling#933]
Hi, im having issues getting the remote GDB session working with one of the UEFI examples.

Here is my test setup:

1. Fresh Install of ubuntu 18 and master branch of Qiling
    sudo apt-get update
    sudo apt-get upgrade
    sudo apt install python3-pip git cmake
    git clone https://github.com/qilingframework/qiling
    cd qiling
    sudo pip3 install . 
    git submodule update --init --recursive

2. Launch EfiPoolFault.efi with qltool and launch gdb server. (this seems to work fine)
    cd qiling/examples/rootfs/x8664_efi/bin
    qltool run -f EfiPoolFault.efi --rootfs . --gdb localhost:9999
 ```
    [=]     Done with loading EfiPoolFault.efi
    [=]     Running from 0x00100280 of EfiPoolFault.efi
    [=]     gdb> Stop at entry point: 0x100280 
    [=]     gdb> Listening on localhost:9999
```
3. Launch GDB and try to connect to the server. Using GDB 8.1
 (gdb) set debug remote 1
(gdb) set remotetimeout 100
(gdb) set architecture i386:x86-64
The target architecture is assumed to be i386:x86-64
(gdb) target remote localhost:9999
Remote debugging using localhost:9999
Sending packet: $qSupported:multiprocess+;swbreak+;hwbreak+;qRelocInsn+;fork-events+;vfork-events+;exec-events+;vContSupported+;QThreadEvents+;no-resumed+;xmlRegisters=i386#6a...Ack
Packet received: PacketSize=3fff;QPassSignals+;QProgramSignals+;QStartupWithShell+;QEnvironmentHexEncoded+;QEnvironmentReset+;QEnvironmentUnset+;QSetWorkingDir+;QCatchSyscalls+;qXfer:libraries-svr4:read+;augmented-libraries-svr4-read+;qXfer:auxv:read+;qXfer:spu:read+;qXfer:spu:write+;
qXfer:siginfo:read+;qXfer:siginfo:write+;qXfer:features:read+;QStartNoAckMode+;qXfer:osdata:read+;multiprocess+;fork-events+;vfork-events+;exec-events+;QNonStop+;QDisableRandomization+;qXfer:threads:read+;ConditionalTracepoints+;TraceStateVaria
Packet qSupported (supported-packets) is supported
Sending packet: $vMustReplyEmpty#3a...Ack
Packet received:
Sending packet: $QStartNoAckMode#b0...Ack
Packet received: OK
Sending packet: $QProgramSignals:0;1;3;4;6;7;8;9;a;b;c;d;e;f;10;11;12;13;14;15;16;17;18;19;1a;1b;1c;1d;1e;1f;20;21;22;23;24;25;26;27;28;29;2a;2b;2c;2d;2e;2f;30;31;32;33;34;35;36;37;38;39;3a;3b;3c;3d;3e;3f;40;41;42;43;44;45;46;47;48;49;4a;4b;4c;4d;4e;4f;50;51;52;53;54;55;56;57;58;59;5a
;5b;5c;5d;5e;5f;60;61;62;63;64;65;66;67;68;69;6a;6b;6c;6d;6e;6f;70;71;72;73;74;75;76;77;78;79;7a;7b;7c;7d;7e;7f;80;81;82;83;84;85;86;87;88;89;8a;8b;8c;8d;8e;8f;90;91;92;93;94;95;96;97;98;99;9a;#a3...Packet received: OK
Sending packet: $Hgp0.0#ad...Packet received: OK
Sending packet: $qXfer:features:read:target.xml:0,fff#7d...Packet received: l<?xml version="1.0"?><!DOCTYPE target SYSTEM "gdb-target.dtd"><target><architecture>i386:x86-64</architecture><osabi>GNU/Linux</osabi><xi:include href="64bit-core.xml"/><xi:include href="64bit-sse.xml"/><xi:i
nclude href="64bit-linux.xml"/><xi:include href="64bit-segments.xml"/><xi:include href="64bit-avx.xml"/><xi:include href="64bit-mpx.xml"/></target>\n
Sending packet: $qXfer:features:read:64bit-core.xml:0,fff#75...Packet received: l<?xml version="1.0"?>\n\n\n<!DOCTYPE feature SYSTEM "gdb-target.dtd">\n<feature name="org.gnu.gdb.i386.core">\n  <flags id="i386_eflags" size="4">\n    <field name="CF" start="0" end="0"/>\n    <field name="" start="1" end="1"/>\n
   <field name="PF" start="2" end=[12 bytes omitted]
Sending packet: $qXfer:features:read:64bit-sse.xml:0,fff#17...Packet received: l<?xml version="1.0"?>\n\n\n<!DOCTYPE feature SYSTEM "gdb-target.dtd">\n<feature name="org.gnu.gdb.i386.sse">\n  <vector id="v4f" type="ieee_single" count="4"/>\n  <vector id="v2d" type="ieee_double" count="2"/>\n  <vector id="v16i8"
 type="int8" count="16"/>\n  <vec[12 bytes omitted]
Sending packet: $qXfer:features:read:64bit-linux.xml:0,fff#fc...Packet received: l<?xml version="1.0"?>\n\n\n<!DOCTYPE feature SYSTEM "gdb-target.dtd">\n<feature name="org.gnu.gdb.i386.linux">\n  <reg name="orig_rax" bitsize="64" type="int" regnum="57"/>\n</feature>\n
Sending packet: $qXfer:features:read:64bit-segments.xml:0,fff#32...Packet received: l<?xml version="1.0"?>\n\n\n<!DOCTYPE feature SYSTEM "gdb-target.dtd">\n<feature name="org.gnu.gdb.i386.segments">\n  <reg name="fs_base" bitsize="64" type="int"/>\n  <reg name="gs_base" bitsize="64" type="int"/>\n</feature>\n
Sending packet: $qXfer:features:read:64bit-avx.xml:0,fff#1b...Packet received: l<?xml version="1.0"?>\n\n\n<!DOCTYPE feature SYSTEM "gdb-target.dtd">\n<feature name="org.gnu.gdb.i386.avx">\n  <reg name="ymm0h" bitsize="128" type="uint128"/>\n  <reg name="ymm1h" bitsize="128" type="uint128"/>\n  <reg name="ymm2h
" bitsize="128" type="uint128"/>[11 bytes omitted]
Sending packet: $qXfer:features:read:64bit-mpx.xml:0,fff#21...Packet received: l<?xml version="1.0"?>\n\n\n<!DOCTYPE feature SYSTEM "gdb-target.dtd">\n<feature name="org.gnu.gdb.i386.mpx">\n  <struct id="br128">\n    <field name="lbound" type="uint64"/>\n    <field name="ubound_raw" type="uint64"/>\n  </struct>
\n\n  <struct id="_bndstatus" size=[14 bytes omitted]
Sending packet: $qXfer:auxv:read::0,1000#6b...Packet received: l!
Sending packet: $QNonStop:0#8c...Packet received: OK
Sending packet: $qTStatus#49...Packet received: T0;tnotrun:0;tframes:0;tcreated:0;tfree:500000;tsize:500000;circular:0;disconn:0;starttime:0;stoptime:0;username:;notes::
Packet qTStatus (trace-status) is supported
Sending packet: $qTfV#81...Packet received: l
Sending packet: $?#3f...Packet received: T0506:0000000000000000;07:e8ffff7700000000;10:8002100000000000;
Sending packet: $qXfer:threads:read::0,fff#03...Packet received: l
Sending packet: $Hc-1#09...Packet received: OK
Sending packet: $qC#b4...Packet received:
Sending packet: $qAttached:a410#bf...Packet received:
Packet qAttached (query-attached) is NOT supported
Sending packet: $qXfer:exec-file:read::0,fff#8a...Packet received: l/home/usr123/qiling/examples/rootfs/x8664_efi/bin/EfiPoolFault.efi
Sending packet: $vFile:setfs:0#bf...Packet received:
Packet vFile:setfs (hostio-setfs) is NOT supported
Sending packet: $vFile:open:6a7573742070726f62696e67,0,1c0#ed...Packet received: F-1
Packet vFile:open (hostio-open) is supported
Reading /home/usr123/qiling/examples/rootfs/x8664_efi/bin/EfiPoolFault.efi from remote target...
warning: File transfers from remote targets can be slow. Use "set sysroot" to access files locally instead.
Sending packet: $vFile:open:2f686f6d652f7573723132332f71696c696e672f6578616d706c65732f726f6f7466732f78383636345f6566692f62696e2f456669506f6f6c4661756c742e656669,0,0#88...Packet received: F-1
warning: "target:/home/usr123/qiling/examples/rootfs/x8664_efi/bin/EfiPoolFault.efi": could not open as an executable file: Unknown error -1.
Reading /home/usr123/qiling/examples/rootfs/x8664_efi/bin/EfiPoolFault.efi from remote target...
Sending packet: $vFile:open:2f686f6d652f7573723132332f71696c696e672f6578616d706c65732f726f6f7466732f78383636345f6566692f62696e2f456669506f6f6c4661756c742e656669,0,0#88...Packet received: F-1
warning: `target:/home/usr123/qiling/examples/rootfs/x8664_efi/bin/EfiPoolFault.efi': can't open to read symbols: Unknown error -1.
Sending packet: $g#67...Packet received: 000000000000000000000000000000000000100000000000000000780000000000000000000000000000000000000000f0ffff7700000000e8ffff770000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
00000000000080021000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
Truncated register 26 in remote 'g' packet

Am I missing something? Sorry I am new to Qiling. Thanks!

**Comment by elicn** [comment]
Hi there and welcome (:
This is indeed a Qiling bug which was already fixed in `dev` branch (see similar bug at #929)

@xwings, can you please push #930 to master?

**Comment by user000543** [comment]
Thanks. I do not get this issue when using the dev branch. 

Closed.
